### PR TITLE
PP-11059 Idle Connection Monitor

### PR DESCRIPTION
**Pull request by james-peacock-gds**
Context: Performance testing of webhooks identified client attempts to re-use expired connections to send webhook messages
- add IdleConnectionMonitor to identify and close idle/expired connections
- add configuration options for IdleConnectionMonitor

### CODE REVIEW

File: src/main/resources/config/config.yaml [hunk 1]
```diff
@@ -93,6 +93,10 @@ webhookMessageSendingQueueProcessorConfig:
   threadDelayInMilliseconds: ${WEBHOOK_MESSAGE_SENDING_QUEUE_THREAD_DELAY_IN_MILLISECONDS:-1000}
   httpClientConnectionPoolSize: ${WEBHOOK_MESSAGE_SENDER_CONNECTION_POOL_SIZE:-20}
   connectionPoolTimeToLive: ${WEBHOOK_MESSAGE_SENDER_CONNECTION_POOL_TIME_TO_LIVE:-60s}
+  connectionPoolIdleConnectionTimeToLive: ${WEBHOOK_MESSAGE_SENDER_CONNECTION_POOL_IDLE_CONNECTION_TIME_TO_LIVE:-5s}
   requestTimeout: ${WEBHOOK_MESSAGE_SENDER_REQUEST_TIMEOUT:-3s}
+  idleConnectionMonitorEnabled: ${WEBHOOK_MESSAGE_SENDER_IDLE_CONNECTION_MONITOR_ENABLED:-true}
```
Comment (kbottla) [inline src/main/resources/config/config.yaml:98]: Would be good to update test-config.yaml too so the connection manager can be enabled in tests and see if there are any issues for it starting with the test config.

Comment (kbottla) [inline src/main/resources/config/config.yaml:98]: Do we want to enable it straightway in production as the default is `true`?

Comment (james-peacock-gds) [reply src/main/resources/config/config.yaml]: done - also added the other required configuration options

Comment (james-peacock-gds) [reply src/main/resources/config/config.yaml]: good point, done.

**Review by kbottla — APPROVED** [review]

**Pull request merged**